### PR TITLE
Add support for Ed25519 sign and verify

### PR DESCRIPTION
**Pull request by kingsleyh**
This PR adds:

1. Ed25519PublicSigningKey
2. Ed25519Signature

To allow signing and verification using the more widely used Ed25519 + SHA512 variant. 

I required this for my http://sushichain.io to allow compatibility with existing javascript ed25519 libraries for the sushichain crypto wallet

### CODE REVIEW

**Comment by kingsleyh** [comment]
I may not need this now - as I'm thinking of generating a javascript version of monocypher using emscripten which means I don't have to worry about compatibility with existing javascript libs

**Comment by kingsleyh** [comment]
Actually I am after all going to need this as the emscripten route is not as nice as the existing js libs - which I've now got working with monocypher.

**Comment by kingsleyh** [comment]
In my branch I'm upgrading monocypher to the latest version.

File: README.md [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+[![Linux CI](https://github.com/Axentro/monocypher.cr/actions/workflows/linux-ci.yml/badge.svg)](https://github.com/Axentro/monocypher.cr/actions/workflows/linux-ci.yml)
+
 Crystal wrapper for a cryptographic library Monocypher ([Official site](https://monocypher.org), [Github page](https://github.com/LoupVaillant/Monocypher))
 
-Note: Sources of Monocypher (version 2.0.5 currently) are included in the shard and will be linked statically with application.
+Note: Sources of Monocypher (version 3.1.2 currently) are included in the shard and will be linked statically with application.
 On the other hand, if libmonocypher.so is present in library paths at the time of compilation, linker seems to prefer dynamic linking.
 
+This repo was modified from the original to add support for the standard ED25519 crytography which Monocypher supplies as an optional component but which was not included in the original Crystal wrapper.
+
```
Comment (konovod) [inline README.md:11]: Should be rephrased to be merged in, well, original repo.

File: README.md [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 ## Installation
 
 1. Add this to your application's `shard.yml`:
 
 ```yaml
 dependencies:
   monocypher:
-    github: konovod/monocypher
+    github: axentro/monocypher
```
Comment (konovod) [inline README.md]: same here.

**Review by konovod — CHANGES_REQUESTED** [review]
LGTM otherwise

**Comment by konovod** [comment]
wow, i'm impressed to see this library used in production. Please modify the readme if you want to merge the PR.

**Comment by kingsleyh** [comment]
Wow got a reply after more than 1 year hahahh! I'll change the readme so you can merge it

**Pull request merged**